src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useContext, useCallback, useState, useEffect } from 'react';
+import { ClientSDKContext } from '../../context/clientSDKContext';
+import { globalSearch, SearchItem, SearchForFilter } from './globalSearch';
+import { Search } from '../common/Search/Search';
+import { PureObject, ApiQuery } from '../..';
+import groupBy from 'lodash/groupBy';
+import styled from 'styled-components';
+import { Icon } from 'antd';
+import { withDefaultTheme } from '../../hoc';
+import { SearchResultList } from './SearchResultList';
+import { CogniteError } from '@cognite/sdk';
+
+class RefCounter {
+  id: number;
+  constructor() {
+    this.id = 0;
+  }
+
+  currentId(): number {
+    return this.id;
+  }
+
+  nextId(): number {
+    return ++this.id;
+  }
+}
+
+export type GlobalSearchStrings = {
+  assets: string;
+  timeSeries: string;
+  events: string;
+  files: string;
+  noDescription: string;
+  emptySearch: string;
+  searchPlaceholder: string;
+};
+
+const defaultStrings: GlobalSearchStrings = {
+  assets: 'Assets',
+  timeSeries: 'Time Series',
+  events: 'Events',
+  files: 'Files',
+  noDescription: '[No Description]',
+  emptySearch: 'Nothing found',
+  searchPlaceholder: 'Search for a resource',
+};
+
+export type RenderSearchItem = (
+  strings: GlobalSearchStrings,
+  item: SearchItem
+) => React.ReactNode;
+export type ItemSelectedCallback = (item: SearchItem) => void;
+
+type GlobalSearchProps = {
+  onError?: (error: CogniteError) => void;
```

Comment:
@polomani is this the correct error type coming from the SDK? I couldn't find any documentation for it.

Reply:
yes!